src/web3/KeychainLock.ts

```diff
@@ -0,0 +1,244 @@
+import {
+  isValidAddress,
+  normalizeAddress,
+  normalizeAddressWith0x,
+  privateKeyToAddress,
+} from '@celo/utils/lib/address'
+import CryptoJS from 'crypto-js'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import { generateKeysFromMnemonic, getStoredMnemonic } from 'src/backup/utils'
+import {
+  listStoredItems,
+  removeStoredItem,
+  retrieveStoredItem,
+  storeItem,
+} from 'src/storage/keychain'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'web3/KeychainLock'
+
+export const ACCOUNT_STORAGE_KEY_PREFIX = 'account--'
+
+export interface KeychainAccount {
+  address: string
+  createdAt: Date
+  importFromMnemonic?: boolean
+}
+
+export interface ImportMnemonicAccount {
+  address: string | null
+  createdAt: Date
+}
+
+// Produces a storage key that looks like this: "account--2022-05-24T13:55:47.117Z--2d936b3ada6142b4248de1847c14fa2f4c5b63c3"
+function accountStorageKey(account: KeychainAccount) {
+  if (!isValidAddress(account.address)) {
+    throw new Error('Expecting valid address for computing storage key')
+  }
+  return `${ACCOUNT_STORAGE_KEY_PREFIX}${account.createdAt.toISOString()}--${normalizeAddress(
+    account.address
+  )}`
+}
+
+async function encryptPrivateKey(privateKey: string, password: string) {
+  return CryptoJS.AES.encrypt(privateKey, password).toString()
+}
+
```

Comment:
Completely copy/pasted. I thought it now made more sense in this file than KeychainSigner